search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -0,0 +1,146 @@
+(ns cmr.search.services.parameters.converters.shapefile
+  "Contains parameter converters for shapefile parameter"
+  (:require 
+    [clojure.java.io :as io]
+    [clojure.string :as str]
+    [cmr.common.config :as cfg :refer [defconfig]]
+    [cmr.common.log :refer (debug)]
+    [cmr.common.mime-types :as mt]
+    [cmr.common-app.services.search.group-query-conditions :as gc]
+    [cmr.common-app.services.search.params :as p]
+    [cmr.common.services.errors :as errors]
+    [cmr.search.models.query :as qm]
+    [cmr.search.services.parameters.converters.geometry :as geo]
+    [cmr.common.util :as util])
+  (:import
+   (java.io BufferedInputStream File FileReader FileOutputStream FileInputStream)
+   (java.nio.file Files)
+   (java.nio.file.attribute FileAttribute)
+   (java.net URL)
+   (java.util HashMap)
+   (java.util.zip ZipFile ZipInputStream)
+   (org.apache.commons.io FilenameUtils)
+   (org.geotools.data DataStoreFinder FileDataStoreFinder Query)
+   (org.geotools.data.simple SimpleFeatureSource)
+   (org.geotools.data.geojson GeoJSONDataStore)
+   (org.geotools.geometry.jts JTS)
+   (org.geotools.referencing CRS)
+   (org.geotools.util URLs)))
+
+(def EPSG-4326-CRS (CRS/decode "EPSG:4326" true))
+
+(defconfig enable-shapefile-parameter-flag
+  "Flag that indicates if we allow spatial searching by shapefile."
+  {:default true :type Boolean})
+
+(defn- unzip-file
+  "Unzip a file (of type File) into a temporary directory and return the directory path as a File"
+  [source]
+  (try
+    (let [target-dir (Files/createTempDirectory "Shapes" (into-array FileAttribute []))]
```

Comment:
I assume this should work fine with Fargate, not positive what they set as the temp directory and how much space there is though. Probably worth sanity checking - we also might need a validation that any passed in shape file when unzipped won't fill up the disk (well we would want that for other reasons too).